File: examples/get.pl
use v5.14;

use open ();
use PerlIO::encoding;
use Encode qw(:fallback_all);
use if (-t STDOUT and $PerlIO::encoding::fallback = HTMLCREF), open => ':locale';
use if (not -t STDOUT), open => ':std', ':encoding(UTF-8)';

use Mojo::UserAgent;

my $url = shift @ARGV or die 'get.pl url opts';
my %opts = map { split /=/, $_, 2 } @ARGV;

my $tx = Mojo::UserAgent->new(%opts)->get($url);
if (my $err = $tx->error) {
    die "$err->{code} response: $err->{message}" if $err->{code};
    die "Connection error: $err->{message}";
}
say $tx->res->text;

File: examples/get.py
import Pyjo.UserAgent

import codecs
import sys

from Pyjo.Util import die

if sys.stdout.isatty():
    sys.stdout = codecs.getwriter(sys.stdout.encoding)(sys.stdout.detach() if hasattr(sys.stdout, 'detach') else sys.stdout, 'xmlcharrefreplace')
else:
    sys.stdout = codecs.getwriter('utf-8')(sys.stdout.detach() if hasattr(sys.stdout, 'detach') else sys.stdout)

try:
    url = sys.argv[1]
except IndexError:
    raise Exception('get.py url opts')

opts = dict(map(lambda a: a.split('='), sys.argv[2:]))

tx = Pyjo.UserAgent.new(**opts).get(url)
err = tx.error
if err:
    if err['code']:
        die("{code} response: {message}".format(**err))
    else:
        die("Connection error: {message}".format(**err))
print(tx.res.text)
